srv/modules/runners/changed.py

```diff
@@ -74,6 +75,34 @@ def dependencies(self):
         """
         return self._depends
 
+    @property
+    def depends(self):
+        """
+        Getter for name attr
+        """
+        return self._depends
+
+    @depends.setter
+    def depends(self, role):
+        """
+        Setter for name attr
+        """
+        self._depends = role
+
+    def _set_depends(self):
+        """ Set dependency to self if not global """
+        if self._role_name == 'global':
+            # if the global config changed
+            # stage restart on the core services
+            self.add_dependencies(Role(role_name='mon'))
+            self.add_dependencies(Role(role_name='mgr'))
+            self.add_dependencies(Role(role_name='storage'))
+            # also stage restarts on gateways (igw is not maintained here)
+            self.add_dependencies(Role(role_name='mds'))
+            self.add_dependencies(Role(role_name='rgw'))
```

Comment:
This causes a loop due to this https://github.com/SUSE/DeepSea/blob/edbb5ad2381332027aac7b9268bbd8c768fbd773/srv/modules/runners/changed.py#L125-L145

I'm not sure why we set the dependency for `global` on a rgw_configuration change?. I'd argue for getting rid of this

Reply:
by the looks of it, we don't need `rgw_configurations` at all. According to 

https://github.com/SUSE/DeepSea/blob/master/srv/salt/ceph/rgw/restart/controlled/default.sls#L1-L10

we restart all `rgw_configurations` if a `restart_rgw` grain is set. No need to check rgw_configrations inside the `changed` runner at all.